discovery/azure/azure_test.go

```diff
@@ -201,3 +277,50 @@ func TestMapFromVMScaleSetVMWithTags(t *testing.T) {
 		t.Errorf("Expected %v got %v", expectedVM, actualVM)
 	}
 }
+
+func TestGetPowerStatusFromVM(t *testing.T) {
```

Comment:
I'd use [table-driven tests](https://dave.cheney.net/2019/05/07/prefer-table-driven-tests) here as they are easier to read/maintain IMO.